src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -46,7 +41,16 @@ internal static PowerShellAssemblyLoadContext InitializeSingleton(string basePat
             lock (s_syncObj)
             {
                 if (Instance != null)
+                {
                     throw new InvalidOperationException(SingletonAlreadyInitialized);
+                }
+
+                // Add last resort native dll resolver.
+                // Default order:
+                //      1. AssemblyLoadContext.LoadUnmanagedDll()
+                //      2. System.Runtime.InteropServices.DllImportResolver callbacks
+                //      3. AssemblyLoadContext.Default.ResolvingUnmanagedDll handlers
+                AssemblyLoadContext.Default.ResolvingUnmanagedDll += NativeDllHandler;
```

Comment:
How about moving this to the `PowerShellAssemblyLoadContext` constructor, next to `AssemblyLoadContext.Default.Resolving += Resolve;`?

Reply:
Done.